experimenter/experimenter/nimbus_ui_new/views.py

```diff
@@ -114,4 +116,21 @@ def get_context_data(self, **kwargs):
             (outcome, f"{doc_base_url}{self.object.application}/{outcome}")
             for outcome in secondary_outcomes
         ]
+        context["qa_edit_mode"] = self.request.GET.get("edit_qa_status") == "true"
+        if context["qa_edit_mode"]:
+            context["form"] = QAStatusForm(instance=self.object)
         return context
+
+    def post(self, request, *args, **kwargs):
+        self.object = self.get_object()
+        form = QAStatusForm(request.POST, instance=self.object)
+        if form.is_valid():
+            form.save()
+            context = self.get_context_data(object=self.object)
+            return render(request, self.template_name, context)
+        else:
+            # if form is not valid, still prepare context data
+            context = self.get_context_data(object=self.object)
+            context["form"] = form
+            context["qa_edit_mode"] = True
+            return render(request, self.template_name, context)
```

Comment:
Oh instead of doing this, I'd suggest either

1) Using the [FormMixin](https://github.com/django/django/blob/main/django/views/generic/edit.py#L13)
2) Make it a totally separate generic view for POSTing too

Reply:
I think I will perfer approach 2 as it will be easy to maintain for each forms